drivers/spi/opentitan/spi.c

```diff
@@ -0,0 +1,727 @@
+//TODO: add licenses
```

Comment:
Copy paste this from one of the other drivers + update to 2025. We can't merge without this!